### ISSUE TITLE
Lack of completion for Quotes Terms

### ISSUE THREAD

**Issue by rochala** [scalameta/metals#3052]
Hi, I found that metals doesn't autocomplete Terms from Quoted. It applies to everything defined in that trait.

**To Reproduce**

1. Insert following code or clone my repository: https://github.com/rochala/metals-completion-bug
```
import scala.quoted._

object Test {
  def test(using Quotes)(str: String) = {
    import quotes.reflect._
    val msg = Expr(str)
    val printHello = '{ print($msg) }
    val tree = printHello.asTerm //tree has no completion
    tree.show //try writing same line
  }
}
```
2. Try to get autocompletion for value tree
3. See no completion

**Expected behavior**
Completion for Terms should be provided.

**Screenshots**

<img width="520" alt="image" src="https://user-images.githubusercontent.com/48657087/130058246-bf0bcddf-dbdd-48a5-b485-3c64d6f605c9.png">

**Installation:**

- Operating system: macOS
- Editor: Visual Studio Code/Vim
- Metals version: v0.10.6-M1

**Additional context**

None

**Search terms**

Quoted, quotes, reflection, completion

**Comment by tgodzik** [comment]
Awesome! Thanks for the reproduction.

**Comment by griggt** [comment]
Upstream PR: https://github.com/lampepfl/dotty/pull/13368

**Comment by dos65** [comment]
Fixed in 3.0.1-RC1
